ONSdigital/dp-observation-importer
# Update healthcheck and dependency manager

### What

* Use new health check library and client libraries for checker functions
* Use new dependency manager go modules instead of go vendor
* Implement new kafka library, `dp-kakfa`
* Implement new log library `log.go/log` instead of `go-ns/log`
* Fix race conditions in unit tests
* Re-order graceful shutdown objects/services, to prevent data loss during processing
* Create an initialisation package to retain state of what services successfully initiated and prevent early closure of application if these failed (calling the health check endpoint will return the state of the application)

### How to review

* Check unit tests pass, run `Make debug`
* Check code changes make sense
* Check healthcheck endpoint returns the expected data, by running the application and run the following command:
`curl localhost:21700/health`

### Who can review

Anyone
